### PR TITLE
More specs for "squiggly" heredoc.

### PR DESCRIPTION
**Pull request by iliabylich**
@eregon Also extracted all heredoc stuff to a separate file. Let me know if it should be reverted.

### CODE REVIEW

File: language/heredoc_spec.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    it 'allows HEREDOC with <<~identifier, interpolated' do
+      SquigglyHeredocSpecs.unquoted.should == "unquoted interpolated\n"
+    end
+
+    it 'allows HEREDOC with <<"identifier", interpolated' do
+      SquigglyHeredocSpecs.doublequoted.should == "doublequoted interpolated\n"
+    end
+
+    it "allows HEREDOC with <<'identifier', interpolated" do
+      SquigglyHeredocSpecs.singlequoted.should == "singlequoted \#{\"interpolated\"}\n"
```
Comment (eregon) [inline language/heredoc_spec.rb]: So this one is actually not interpolated, right? Could you fix the description?

File: language/heredoc_spec.rb [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    end
+
+    it "selects a least-indented line and removes its indentation from all the lines" do
```
Comment (eregon) [inline language/heredoc_spec.rb]: selects *the* least

**Comment by eregon** [comment]
> Also extracted all heredoc stuff to a separate file.

Great idea!

Thanks for these very nice specs!

**Comment by iliabylich** [comment]
Fixed, good catch. Any other comments?

**Comment by eregon** [comment]
No, let's merge it! :smiley:

**Pull request merged**